### ISSUE TITLE
index.html added to chartmuseum URL

### ISSUE THREAD

**Issue by sauber** [rancher/rancher#22881]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Apps -> Add Catalog -> Catalog URL -> Create
Write a URL Like: http://192.168.62.169:9580/api/chartrepo/louvre/charts.html

**Result:**
2019/09/16 10:48:51 [ERROR] CatalogController louvre [catalog] failed with : Unexpected HTTP status code 404 from [http://192.168.62.169:9580/api/chartrepo/louvre/charts.html/index.yaml], expected 200

Rancher has added a /index.html to URL, which results in 404.

**Other details that may be helpful:**
Same results with a dir-style URL: http://192.168.62.169:9580/api/chartrepo/louvre/charts

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA): Single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
(paste the output here)
```

- Docker version (use `docker version`):

```
Server: Docker Engine - Community
 Engine:
  Version:          19.03.2
  API version:      1.40 (minimum version 1.12)```

**Comment by sauber** [comment]
I suspect that the issue is caused here:
https://github.com/rancher/rancher/blob/master/pkg/catalog/helm/helm.go#L112 :

```
func (h *Helm) downloadIndex(indexURL string) (*RepoIndex, error) {
	indexURL = strings.TrimSuffix(indexURL, "/")
	indexURL = indexURL + "/index.yaml"
	resp, err := h.request(indexURL)
```

**Comment by sauber** [comment]
I have created Pull Request https://github.com/rancher/rancher/pull/23561 to resolve the issue.

**Comment by TM-Developers** [comment]
hi @sauber , I see this hasn't been pushed yet to any rancher version. Is there a workaround? I'm using rancher 2.3.4

**Comment by sauber** [comment]
There is no workaround.

**Comment by TeckTn** [comment]
Hi, Any fix for this issue ? I am using the v2.3.2 and I can't add an Azure DevOps Repo as a repo catalog.
